tests/async/tasyncrecursion.nim

```diff
@@ -17,5 +17,6 @@ proc asyncRecursionTest*(): Future[int] {.async.} =
     inc(i)
 
 when isMainModule:
+  setGlobalDispatcher(newDispatcher())
```

Comment:
Why is this necessary? The main thread should always have an implicit dispatcher.